src/image.c

```diff
@@ -2651,11 +2651,54 @@ GdipGetEncoderParameterList (GpImage *image, GDIPCONST CLSID *clsidEncoder, UINT
 GpStatus WINGDIPAPI
 GdipGetImageThumbnail (GpImage *image, UINT thumbWidth, UINT thumbHeight, GpImage **thumbImage, GetThumbnailImageAbort callback, VOID *callbackData)
 {
+	GpStatus status;
+	PixelFormat format;
+	GpImage *result;
+	GpGraphics *graphics;
+
 	if (!image || !thumbImage)
 		return InvalidParameter;
 
-	/* We don't really need to implement this; we've got it handled in System.Drawing directly */
-	return NotImplemented;
+	if ((thumbWidth && !thumbHeight) || (!thumbWidth && thumbHeight))
+		return OutOfMemory;
+
+	if (!thumbWidth || !thumbHeight)
+		thumbWidth = thumbHeight = 120;
+
+	if (!thumbHeight)
+		thumbHeight = 120;
```

Comment:
This is already handled by the previous `if`.

Reply:
Maybe something like this would be easier to read?

```
if (!thumbWidth && !thumbHeight) {
  thumbWidth = thumbHeight = 120;
} else if (!thumbWidth || !thumbHeight) {
  return OutOfMemory;
}
```